parallel.go

```diff
@@ -289,20 +289,59 @@ func (r Parallel) forKey(key string) Parallel {
 	})
 }
 
+// mergeQueryEvents limits `routing.QueryError` events to only be sent on the context in case all parallel
+// routers fail.
+func (r Parallel) mergeQueryEvents(ctx context.Context) (context.Context, context.CancelFunc) {
+	subCtx, cancel := context.WithCancel(ctx)
+	subCtx, evCh := routing.RegisterForQueryEvents(subCtx)
+	go func() {
+		var errEvt *routing.QueryEvent
+		successfulEvent := false
+		for {
+			select {
+			case <-ctx.Done():
```

Comment:
I'd still remove this. That way we read the _last_ event from the event channel.

Reply:
we're already in a closing race if the outer Ctx is done. I don't think there's an ordering guarantee that the event channel on the parent context that this loop writes to will stay alive longer than the sub-context's channel.

Reply:
Ah, ok, this is the _outer_ ctx. I'm still not sure if this is necessary, given that we cancel the inner ctx, but it should be fine.